### ISSUE TITLE
v0.21.0 mirror hosted by proxy.golang.org is broken

### ISSUE THREAD

**Issue by sarisia** [bwmarrin/discordgo#783]
Go Module Mirror (proxy.golang.org) seems mirroring broken release, so `go get` fails:
```
$ go get -u github.com/bwmarrin/discordgo
go: downloading github.com/bwmarrin/discordgo v0.21.0
go: downloading golang.org/x/crypto v0.0.0-20200604202706-70a84ac30bf9
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading golang.org/x/sys v0.0.0-20200620081246-981b61492c35
go: github.com/gorilla/websocket upgrade => v1.4.2
go: golang.org/x/crypto upgrade => v0.0.0-20200604202706-70a84ac30bf9
go: golang.org/x/sys upgrade => v0.0.0-20200620081246-981b61492c35
# github.com/bwmarrin/discordgo
../../../../go/pkg/mod/github.com/bwmarrin/discordgo@v0.21.0/wsapi.go:852:19: (*Session).Close redeclared in this block
        previous declaration at ../../../../go/pkg/mod/github.com/bwmarrin/discordgo@v0.21.0/wsapi.go:846:6
```

# workaround

Do not use module mirror:
```
$ go clean --modcache
$ GOPROXY=direct go get github.com/bwmarrin/discordgo@v0.21.0
go: downloading github.com/bwmarrin/discordgo v0.21.0
go get: github.com/bwmarrin/discordgo@v0.21.0: verifying module: checksum mismatch
        downloaded: h1:a4V4v2IPHPy7l5XVbjJkJAj9R2Lhvz7vs5I4Mq3OFYk=
        sum.golang.org: h1:jGuwVZTUHZBUFZ3sm5cOqrwphGQWeL0/9XkaCbDEcrs=

SECURITY ERROR
This download does NOT match the one reported by the checksum server.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.

For more information, see 'go help module-auth'.
```

# to solve...

As described in https://proxy.golang.org , it seems there's no way to remove package once mirrored:
> I removed a bad release from my repository but it still appears in the mirror, what should I do?
Whenever possible, the mirror aims to cache content in order to avoid breaking builds for people that depend on your package, so this bad release may still be available in the mirror even if it is not available at the origin. The same situation applies if you delete your entire repository. We suggest creating a new version and encouraging people to use that one instead.

Please consider release newer version.

**Comment by CarsonHoffman** [comment]
Thank you for the detailed explanation and solution. We had cherry-picked a few bug-fix commits onto `master` during the `v0.20` cycle, and unfortunately since I didn't rebase `develop` onto `master` before merging one of those changes ended up being played twice in the merge.

I've [released v0.21.1](https://github.com/bwmarrin/discordgo/commit/eb3514551b4e8a261f880ea6dae2b7efac9f5ade); this should fix this issue. I am able to `go get` the package locally, but feel free to re-open this issue if this didn't successfully fix it.

Again, thank you for the detailed report, and sorry for the trouble!

**Comment by sarisia** [comment]
Issue resolved. Thank you for quick respond!
